main.cc

```diff
@@ -121,9 +123,12 @@ class Matrix_t : public repl_new
     // localalloc a single row
     void allocateRow(Index_t i)
     {
-        for (Index_t j = 0; j < nrows_per_nodelet_; ++j)
+        for (Index_t row_idx= 0; row_idx < nrows_; ++row_idx)
```

Comment:
I think this is incorrect. This function is being spawned on each nodelet. How many `new`'s will you want to do on each nodelet?

Reply:
From Eric's comment on bug 60: _"Assuming you will have multiple rows per nodelet, you should spawn one thread per nodelet and have each thread allocate all of the local rows."_ 

So the first for loop stays NODELETS(), the local alloc for loop will be nrows_per_nodelet_ 
fixed in the next commit